### DISCUSSION TITLE
OAuth2 without email address

### DISCUSSION THREAD

**Discussion by CPSibo** [requarks/wiki#6225] [Help / Questions] [answered]
I'm trying to get the Generic OAuth2 provider working. Unfortunately, the SSO service I'm hooking into doesn't provide email addresses, only usernames. It seems like this shouldn't be a problem but user accounts seem like they can _only_ be made with a valid email. Whether I try to create the user ahead of time or allow self-registration, both complain that the username isn't a valid email.

Is there any way to turn this off?

Thank you.

**Comment by NGPixel** [comment, marked as answer]
No, Wiki.js requires an email address for all users. You could however edit the oauth2 module to automatically append a fake domain to the username to fill in the email field.
